app/models/activity.rb

```diff
@@ -0,0 +1,58 @@
+class Activity < ActiveRecord::Base
+  before_create { |activity| activity.event = generate_key(activity.object, activity.event) }
+
+  belongs_to :object, polymorphic: true
+  belongs_to :actor, inverse_of: :activities, class_name: User.name
+  has_many :course_notifications, inverse_of: :activity, dependent: :destroy
```

Comment:
Another thing I wanted you to consider was, is placing both tables together and using a polymorphic `:to` column more advantageous?